tests/durability/test_disk_failure_recovery.py

```diff
@@ -265,32 +249,112 @@ def test_sns_repair_fail_disk_equal_to_k(self):
+                                                    skipcleanup=True)
         assert_utils.assert_true(resp[0], resp[1])
-        pod_name = resp[1]
 
-        for cnt in range(disk_fail_cnt):
-            selected_disk = random.choice(all_disks)  # nosec
-            fail_disk = selected_disk.split('$')
-            LOGGER.info("disk fail loop: %s, disk selected for failure: %s", cnt + 1, fail_disk)
-            resp = self.dsk_rec_obj.change_disk_status_hctl(self.node_master_list[0], pod_name,
-                                                            fail_disk[0], fail_disk[2], "failed")
-            LOGGER.info("fail disk resp: %s", resp)
-            self.failed_disks.append(selected_disk)
+        LOGGER.info("Step 6: Check cluster status")
+        resp = self.hlth_master_list[0].all_cluster_services_online()
+        assert_utils.assert_true(resp[0], resp[1])
 
+        LOGGER.info("Step 7: Start SNS repair")
+        resp = self.dsk_rec_obj.sns_repair(self.node_master_list[0], "start", self.pod_name)
+        LOGGER.info("sns start resp: %s", resp)
         time.sleep(self.delay_sns_repair)
+
+        LOGGER.info("Step 8: Check degraded byte count after disk fail")
+        degraded_byte_cnt_after_repair = self.dsk_rec_obj.get_byte_count_hctl \
+            (self.hlth_master_list[0], "degraded_byte_count")
+        LOGGER.info("degraded byte cunt: %s", degraded_byte_cnt_after_repair)
+
+        if degraded_byte_cnt_after_repair >= degraded_byte_cnt_after_fail:
+            assert_utils.assert_true(False, "Degraded byte count after sns repair is greater than "
+                                            "or equal to degraded byte count after disk fail")
+        else:
+            LOGGER.info("Degraded byte count after sns repair is less than "
+                        "degraded byte count after disk fail")
+
+        LOGGER.info("Step 9: Check cluster status after repair")
+        resp = self.hlth_master_list[0].all_cluster_services_online()
+        assert_utils.assert_true(resp[0], resp[1])
+
+        LOGGER.info("Step 10: Read data written in step 1")
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    log_prefix=self.test_prefix[0],
+                                                    skipwrite=True, skipcleanup=True)
+        assert_utils.assert_true(resp[0], resp[1])
+
+        LOGGER.info("Step 11: Do IOs(Write and Read)")
+        self.test_prefix.append('test-36581-after-recovery')
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    log_prefix=self.test_prefix[-1],
+                                                    skipcleanup=True)
+        assert_utils.assert_true(resp[0], resp[1])
+
+        LOGGER.info("COMPLETED: Test SNS repair works fine with failed disks "
+                    "are equal to K(parity units)")
+
+    @pytest.mark.lc
```

Comment:
might need another marker for durability or something

Reply:
ok